macOS/DuckDuckGoVPN/VPNProxyLauncher.swift

```diff
@@ -58,7 +58,11 @@ final class VPNProxyLauncher {
 
     private func statusChanged(notification: Notification) {
         Task { @MainActor in
-            let isProxyConnectionStatusChange = await proxyController.connection == notification.object as? NEVPNConnection
+            guard let connection = notification.object as? NEVPNConnection else {
+                return
+            }
+
+            let isProxyConnectionStatusChange = proxyController.isUsingConnection(connection)
```

Comment:
This is the core fix for the loop.  The issue was caused because `proxyController.connection` causes our proxy configuration to be loaded if it has not been created before.  Loading the configuration fires a status update (thanks Apple).

Reply:
Note: I might need to push this to today's release.